test/2fa.test.js

```diff
@@ -202,94 +193,163 @@ describe('2fa method management', function () {
+                    return testAccountHelper.verify2faMethod({
+                        accountId,
+                        requestId: REQUEST_ID_FOR_INITIALIZING_2FA,
+                        securityCode
+                    })
+                        .then(expectFailedWithCode(401, '2fa code expired'));
+                });
             });
 
-            it('securityCode that is more than 5 minutes old should fail', async () => {
-                return testAccountHelper.verify2faMethod({
-                    accountId,
-                    requestId: REQUEST_ID_FOR_INITIALIZING_2FA,
-                    securityCode
-                })
-                    .then(expectFailedWithCode(401, '2fa code expired'));
+            describe('malformed security codes should be rejected', () => {
+                const expectInvalid2faCodeProvidedError = expectFailedWithCode(401, 'invalid 2fa code provided');
+
+                it('verify 2fa method should fail when given code that is too long', async () => {
+                    return testAccountHelper.verify2faMethod({
+                        accountId,
+                        requestId: REQUEST_ID_FOR_INITIALIZING_2FA,
+                        securityCode: '1234567',
+
+                    })
+                        .then(expectInvalid2faCodeProvidedError);
+                });
+
+                it('verify 2fa method should fail when given code that is too short', async () => {
+                    return testAccountHelper.verify2faMethod({
+                        accountId,
+                        requestId: REQUEST_ID_FOR_INITIALIZING_2FA,
+                        securityCode: '123',
+
+                    })
+                        .then(expectInvalid2faCodeProvidedError);
+                });
+
+                it('verify 2fa method should fail when given code that is not numeric', async () => {
+                    return testAccountHelper.verify2faMethod({
+                        accountId,
+                        requestId: REQUEST_ID_FOR_INITIALIZING_2FA,
+                        securityCode: 'dsfasdsgah3y',
+                    })
+                        .then(expectInvalid2faCodeProvidedError);
+                });
+
+                it('verify 2fa method should fail when no code is provided', async () => {
+                    return testAccountHelper.verify2faMethod({
+                        accountId,
+                        requestId: REQUEST_ID_FOR_INITIALIZING_2FA,
+                    })
+                        .then(expectInvalid2faCodeProvidedError);
+                });
             });
         });
 
-        describe('malformed security codes should be rejected', () => {
-            const expectInvalid2faCodeProvidedError = expectFailedWithCode(401, 'invalid 2fa code provided');
-            const expectInvalid2faCodeForRequestError = expectFailedWithCode(401, '2fa code not valid for request id');
+        describe('email', () => {
```

Comment:
Above changes are just rearrangements, real changes are from here ⬇️ . Now that `2fa-email` and `2fa-phone` have different verification paths i've added tests to cover both.